backends/bmv2/common/control.h

```diff
@@ -444,6 +444,8 @@ class ControlConverter : public Inspector {
             for (auto k : keyset->components) {
                 auto key = new Util::JsonObject();
                 auto tableKey = table->getKey()->keyElements.at(keyIndex);
+                BUG_CHECK(!tableKey->expression->type->is<IR::Type_Unknown>(),
```

Comment:
Why do you use the `type` field instead of looking up the type in the typeMap? That one is most certainly accurate.